src/URI.js

```diff
@@ -1223,6 +1227,26 @@
   };
 
   // compound accessors
+  p.origin = function(v, build) {
+    var parts;
+
+    if (this._parts.urn) {
+      return v === undefined ? '' : this;
+    }
+
+    if (v === undefined) {
+      return this._parts.hostname ? URI.buildOrigin(this._parts) : '';
```

Comment:
I think I would not have added `.buildOrigin()` but rather used the already existing accessors:

```js
return [this.protocol(), this.authority()].join('://');
```